src/Eagle.ts

```diff
@@ -4496,7 +4496,7 @@ export class Eagle {
         }
     }
 
-    addEdge = async (srcNode: Node, srcPort: Field, destNode: Node, destPort: Field, loopAware: boolean, closesLoop: boolean): Promise<Edge> => {
+    addEdge = async (srcNode: Node, srcPort: Field, destNode: Node, destPort: Field, loopAware: boolean, closesLoop: boolean, addNodeAndConnect: boolean): Promise<Edge> => {
```

Comment:
I think "addNodeAndConnect" might be a confusing name for this argument. Since the addEdge function does not "addNode" or "Connect" based on the value of this variable. In other words, it refers to context outside the function itself.

Maybe calling it something based on what it does would be better. Something like "renameNodeAndPorts" or "renameDataNodeAndPorts"?